BlueprintUI/Sources/Internal/CacheTree.swift

```diff
@@ -0,0 +1,52 @@
+import CoreGraphics
+import os.log
+
+/// A size cache that also holds subcaches.
+protocol CacheTree: AnyObject {
```

Comment:
Considered renaming this to something more generalized, like `RenderContext`

Reply:
This seems good to me – I would've maybe gone with `MeasurementCache` or something